iothub/service/design.md

```diff
@@ -0,0 +1,119 @@
+# RBAC API Design for IoT hub
+
+## Service Client
```

Comment:
Should we remove this file? My impression is that we used them during design, and then removed them after impl. We wouldn't want to have to keep them in sync with impl long term, right?

Reply:
I'd like to see this removed as well

Reply:
yes i'll remove it.